superset/databases/utils.py

```diff
@@ -51,30 +62,27 @@ def get_col_type(col: dict[Any, Any]) -> str:
     return dtype
 
 
-def get_table_metadata(
-    database: Any, table_name: str, schema_name: Optional[str]
-) -> dict[str, Any]:
+def get_table_metadata(database: Any, table: Table) -> TableMetadataResponse:
```

Comment:
should the database type be a Database model?

Reply:
Yeah, good catch!